apps/mobile-wallet/src/features/modals/AppModals.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import styled from 'styled-components/native'
+
+import BuyModal from '~/features/buy/BuyModal'
+import { selectAllModals } from '~/features/modals/modalSelectors'
+import { useAppSelector } from '~/hooks/redux'
+
+const AppModals = () => {
+  const openedModals = useAppSelector(selectAllModals)
+
+  return (
+    <ModalsContainer pointerEvents={openedModals.length > 0 ? 'auto' : 'none'}>
+      {openedModals.map((modal) => {
+        switch (modal.params.name) {
+          case 'BuyModal':
+            return <BuyModal id={modal.id} key={modal.id} />
+        }
+      })}
+    </ModalsContainer>
+  )
+}
```

Comment:
Do we even want to render anything if `openedModals.length === 0`?

```ts
if (openedModals.length === 0) return null
```

Reply:
Done